src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/transforms/utils.ts

```diff
@@ -135,7 +152,7 @@ export const buildDatasetState = (
   }
 
   const reference = (references ?? []).find(
-    (ref) => ref.name === `indexpattern-datasource-${layerId}`
+    (ref) => ref.name === `indexpattern-datasource-layer-${layerId}`
```

Comment:
IN general these prefixes needs to be moved somewhere and reused correctly. This is part of the architecture, so is necessary to reuse it